.claude/skills/layer5-blog-writer/scripts/rasterize.py

```
@@ -0,0 +1,160 @@
+
+        try:
+            from PIL import Image
+        except ImportError:
+            fallback = out_path.with_suffix(".png")
+            shutil.move(str(shot), fallback)
+            raise RasterizeError(
+                f"Pillow is not installed, so {suffix} encoding is unavailable. "
+                f"Wrote {fallback.name} instead, which is roughly 6x larger than "
+                f"the intended JPEG. Install Pillow (pip install pillow) and rerun "
+                f"to ship the smaller file."
+            )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)